homeassistant/components/sensor/wink.py

```diff
@@ -38,11 +38,39 @@ class WinkSensorDevice(Entity):
 
     def __init__(self, wink):
         self.wink = wink
+        self._unit_of_measurement = self.wink.UNIT
 
     @property
     def state(self):
         """ Returns the state. """
-        return STATE_OPEN if self.is_open else STATE_CLOSED
+        capability = self.wink.capability()
+        if capability == "humidity":
+            state = self.wink.humidity_percentage()
+        elif capability == "temperature":
+            state = self.wink.temperature_float()
+        elif capability == "loudness":
```

Comment:
Yeah, probably. There are several wink sensors that should probably be
moved over to binary sensors. I will look into it.
On Feb 22, 2016 9:46 PM, "Paulus Schoutsen" <notifications@github.com>
wrote:

> In homeassistant/components/sensor/wink.py
> <https://github.com/balloob/home-assistant/pull/1377#discussion_r53730067>
> :
>
> >
> >      @property
> >      def state(self):
> >          """ Returns the state. """
> > -        return STATE_OPEN if self.is_open else STATE_CLOSED
> > +        capability = self.wink.capability()
> > +        if capability == "humidity":
> > +            state = self.wink.humidity_percentage()
> > +        elif capability == "temperature":
> > +            state = self.wink.temperature_float()
> > +        elif capability == "loudness":
>
> Shouldn't these be exposed as binary sensors?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/balloob/home-assistant/pull/1377/files#r53730067>.
>